frame/thread/bli_l3_decor_openmp.c

```diff
@@ -241,6 +248,27 @@ void bli_l3_thread_decorator_thread_check
 
 		// Synchronize all threads and continue.
 		_Pragma( "omp barrier" )
+
+                return;
+	}
+
+	// Check if the number of OpenMP threads created within this parallel
+	// region is different from the number of threads that are available
+	// to BLIS in the calling context.
+	if ( n_threads_hwmask < n_threads || n_threads_hwmask < n_threads_real)
+	{
+                bli_print_msg( "The affinity mask on this process does not have "
+                               "enough HW threads for your requested SW threads.",
+                               __FILE__, __LINE__ );
+                bli_abort();
+
+		bli_thrcomm_init( n_threads_hwmask, gl_comm );
+		bli_rntm_set_num_threads_only( n_threads_hwmask, rntm );
+#warning HELP ME HERE
+		bli_rntm_set_ways_only( 1, 1, 1, 1, 1, rntm );
```

Comment:
@devinamatthews this is the part where i need help. i can just do serialization, but i was thinking about trying to do slightly better, e.g. if i have 40 cores and the user wants to run on 80 threads, i can set things up to use 40 threads properly.

Reply:
Reducing the number of active threads after thread creation is possible, but not easy. If this logic can go before the OpenMP region then things are much easier.